src/v/cloud_storage/cache_service.cc

```diff
@@ -973,15 +971,30 @@ ss::future<> cache::start() {
             });
         });
         _tracker_timer.arm_periodic(access_timer_period);
+
+        _tracker_sync_timer.set_callback([this] {
+            ssx::spawn_with_gate(_gate, [this]() -> ss::future<> {
+                return sync_access_time_tracker().handle_exception(
+                  [](auto eptr) {
+                      vlog(
+                        cst_log.error,
+                        "failed to sync access time tracker: {}",
+                        eptr);
+                  });
+            });
+        });
+        _tracker_sync_timer.arm(tracker_sync_period);
```

Comment:
So the sync will be done every 6 hours in a non-blocking manner.
Maybe we should sync early if the tracker is empty (this could happen when we upgrade) or when we're low on space but tracker says that we should have space? WDYT?

Reply:
The sync is based on `access_time_tracker::trim` which is one-way right now, it will only ever remove items from the cache and never add new items, which originally made sense because it was expected to remove manually deleted files from the tracker.

Right now the only way to add entries to the tracker is to issue a `get` on the cache. 

To populate an empty tracker, maybe the sync (and tracker's trim method) should have an optional flag to enable it to add items in addition to  removing them, IE when the flag argument is off by default but when it is on, if the item is present in incoming set but not in tracker, then it is added. If the item is not in the incoming set but present in tracker, then it is removed.

Reply:
> when we're low on space but tracker says that we should have space

Would this happen because of extra tmp files and the index and tx files which are present on disk but not in the tracker? 

Even if we sync the tracker state with disk in an additive manner, we would probably skip adding these files to the tracker, so the size difference might still be there after the sync.

Reply:
> To populate an empty tracker, maybe the sync (and tracker's trim method) should have an optional flag to enable it to add items in addition to removing them

maybe the tracker could have a field which is populated when the tracker is loaded, if the tracker was upgraded from v1 to v2 and lost all data then we should set the flag. And if the flag is set we could start the sync early.

Reply:
> Would this happen because of extra tmp files and the index and tx files which are present on disk but not in the tracker?

because of this or because redpanda added some files to the cache and crashed before the tracker was saved to disk

Reply:
On startup in `cleanup_at_start` we sync the tracker. I added a flag there to do add entries to tracker if they are found in directory but missing in the data structure (except index, tx and tmp files). I think this should take care of both scenarios:

1. empty tracker due to upgrade - the sync is done after read from disk so we will populate the tracker during sync with on-disk data
2. tracker not written during shutdown - the sync will write missing entries into the tracker on startup